pkg/cri/resource-manager/cache/container.go

```diff
@@ -111,6 +111,18 @@ func (c *container) fromCreateRequest(req *cri.CreateContainerRequest) error {
 
 	c.TopologyHints = topology.MergeTopologyHints(c.TopologyHints, getKubeletHint(c.GetCpusetCpus(), c.GetCpusetMems()))
 
+	class, ok := c.GetEffectiveAnnotation(RDTClassKey)
+	if !ok {
+		class = string(c.GetQOSClass())
+	}
+	c.SetRDTClass(class)
```

Comment:
Just a thought: this functionality is duplicated in `fromListResponse(...)` so it could be split out into a separate method `setInitialRDTClass()`. And same thing for the blockio class

Reply:
Updated accordingly.